src/pallene/pallenelib.lua

```diff
@@ -43,9 +43,46 @@ return [==[
 #include <stdarg.h>
 #include <locale.h>
 #include <math.h>
+#include <stdbool.h>
 
 #define PALLENE_UNREACHABLE __builtin_unreachable()
 
+/* Part of Pallene Tracer. */
+#define PALLENE_FRAMEENTER(L, name, sig) pt_frame_t _frame = {                                  \
+                                             .fn_name = PALLENE_SOURCE_FILE_WO_EXT "." name,    \
+                                             .mod_name = PALLENE_SOURCE_FILE,                   \
+                                             .line = -1                                         \
+                                         };                                                     \
+                                        pallene_tracer_frameenter(L, &_frame, sig)
+#define PALLENE_GLOBAL_SETLINE(L, line) pallene_tracer_global_setline(L, line)
+#define PALLENE_SETLINE(line)           pallene_tracer_setline(&_frame, line)
+#define PALLENE_FRAMEEXIT(L, ...)       pallene_tracer_frameexit(L);                            \
+                                        return __VA_ARGS__
+
+/* Pallene Tracer related data-structures. */
+typedef struct pt_frame {
+    /* Max name length is 127. */
+    char fn_name[128];
+    char mod_name[128];
```

Comment:
256 bytes is a lot of overhead. Would it be possible to allocate these names statically? That would also get rid of the length limit.

Reply:
Yeah, I think we can allocate them statically and store pointers here.

Reply:
Yeah, it's doable.